native/screens/SignIn.jsx

```diff
@@ -1,103 +1,166 @@
-// import { useEffect, useState } from "react";
-// import { useNavigate } from "react-router-dom";
-// // assets
-// import { useDispatch, useSelector } from "react-redux";
-// import { loginUser } from "../../services/redux/slices/uiActions.js";
-// import "../SignUp/SignUp.scss";
-// import { handleChange } from "../../services/utils/formHandlers";
-// import { SingleActionButton } from "../components/Button/SingleActionButton";
-// import { checkEmailAuth, signOut, verify } from "../../services/api/users";
-import { StyleSheet, Text, View } from "react-native";
+import { useState, createRef, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { signOut, checkEmailAuth, verify } from '../services/api/users.js';
+import { loginUser } from '../services/redux/actions/uiActions.js';
+
+// Native Components
+import { 
+  StyleSheet, 
+  Text, 
+  View, 
+  TextInput, 
+  TouchableOpacity, 
+  Button,
+  Alert,
+} from "react-native";
 
-export const SignIn = () => {
-  return (
-    <View>
-      <Text>SignIn</Text>
-    </View>
-  );
-};
 
-// export const ReactSignIn = () => {
-//   const navigate = useNavigate();
-//   const dispatch = useDispatch();
-//   const { _id: userId } = useSelector((state) => state.ui.user);
-//   const [authError, setAuthError] = useState(null);
-//   const [noAccountError, setNoAccountError] = useState(null);
-//   const [loginInfo, setLoginInfo] = useState({
-//     email: "test@test.com",
-//     password: "test",
-//   });
+export const SignIn = ({navigation}) => {
+  // redux
+  const dispatch = useDispatch();
+  // error states
+  const [noAccountError, setNoAccountError] = useState(null);
+  const [authError, setAuthError] = useState(null);
+  // log in state
+  const [logInInfo, setLogInInfo] = useState({
+    email: '',
+    password: '',
+  });
+  // input references
+  const { emailInputRef, passwordInputRef } = createRef();
 
-//   useEffect(() => {
-//     signOut();
-//   }, []);
+  // sign out user on sign in screen render
+  useEffect(() => {
+    signOut();
+  }, []);
 
-//   const handleSignIn = async (event) => {
-//     event.preventDefault();
-//     await dispatch(loginUser(loginInfo));
-//     const user = await verify();
-//     if (user.email === loginInfo.email) {
-//       navigate(`/users/${userId}/edit`);
-//     } else {
-//       setAuthError(
-//         "Invalid credentials. Please check your details and try again."
-//       );
-//       setLoginInfo((prevState) => {
-//         return {
-//           ...prevState,
-//           password: "",
-//         };
-//       });
-//     }
-//   };
+  const handleSignIn = async () => {
+    await dispatch(loginUser(logInInfo));
+    const user = await verify();
+    if (user.email === logInInfo.email) {
+      setLogInInfo({
```

Comment:
That makes sense, I added this in thinking more short term/ the current state of the app so that when returning to the sign in screen for testing purposes, the form is reset. 

But like you mentioned earlier, ideally a user will not even see/return to the Sign In screen unless they explicitly log out. I'll remove this now and we can revisit where this form reset logic will live according to logout and sign in screen redirect etc.